src/game/net.rs

```diff
@@ -106,18 +194,120 @@ impl<Game: IGame, const CPU: bool> TwoHeadedNetBase<Game, CPU> {
         pos: &Game::Position,
         to_planes: &impl Fn(&Game::Position) -> Vec<Game::Bitboard>,
     ) -> (f32, Vec<(Game::Move, f32)>) {
-        let planes = to_planes(pos);
-        let input = planes_to_tensor::<Game, CPU>(planes);
-        let (val, move_scores) = self.run_net(input);
+        /* Add a new sample to the current batch */
+        let batch;
+        let batch_idx;
+        let sample_idx;
+        let mut run_net = {

```

Comment:
Ideally the batching code shouldn't be under TwoHeadedBaseNet, but realistically it's the only relevant net so not really a problem